### PR TITLE
feat(mui): create refine themes and themed layout

### PR DESCRIPTION
**Pull request by alicanerdurmaz**
following changes implemented for `@refinedev/mui`:

-   blue, purple, magenta, red, orange, yellow, and green themes were added.

-   `<ThemedLayout>`, `ThemedSider`, `<ThemedTitle>`, `<ThemedHeader>` created to use theme colors.

### CODE REVIEW

File: examples/theme-antd-demo/package.json [hunk 1]
```diff
@@ -22,7 +22,7 @@
     "@types/react-dom": "^18.0.0"
   },
   "scripts": {
-    "start": "FAST_REFRESH=false DISABLE_ESLINT_PLUGIN=true react-scripts start",
+    "start": "PORT=3010 FAST_REFRESH=false DISABLE_ESLINT_PLUGIN=true react-scripts start",
```
Comment (alicanerdurmaz) [inline examples/theme-antd-demo/package.json:25]:
```suggestion
    "start": "FAST_REFRESH=false DISABLE_ESLINT_PLUGIN=true react-scripts start",
```

File: .changeset/orange-rules-clean.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+/>
+```
+
+-   `<ThemedLayout>`, `<ThemedSider>`, `<ThemedTitle>`, `<ThemedHeader>` created to use theme colors.
+
+-   `<EditButton>` in `<Show>` type changed to `primary`.
+-   `<CreateButton>` type changed to `primary`.
+
+-   `<AuthPage>` component uses colors from the theme.
+-   `<AuthPageTitle>` added to `AuthPage`
```
Comment (omeraplak) [inline .changeset/orange-rules-clean.md:61]:
```suggestion
-   `<ThemedTitle>` added to `AuthPage`
```

File: examples/theme-mui-demo/src/App.tsx [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+import { BrowserRouter, Routes, Route, Outlet } from "react-router-dom";
+import GitHubIcon from "@mui/icons-material/GitHub";
+import GoogleIcon from "@mui/icons-material/Google";
+
+import { PostList, PostCreate, PostEdit } from "pages/posts";
+import DashboardPage from "pages/dashboard";
+import { authProvider } from "authProvider";
+import { RememeberMe } from "components/remember-me";
+import { useState } from "react";
+import { Stack } from "@mui/system";
```
Comment (salihozdemir) [inline examples/theme-mui-demo/src/App.tsx:37]: Can you fix import order 🙏

File: examples/theme-mui-demo/src/App.tsx [hunk 4]
```diff
@@ -0,0 +1,289 @@
...
+
+const App: React.FC = () => {
+    const [customTheme, setCustomTheme] = useState(RefineThemes.MagentaDark);
+    const [themeSettingsModalIsOpen, setThemeSettingsModalIsOpen] =
+        useState(false);
```
Comment (salihozdemir) [inline examples/theme-mui-demo/src/App.tsx:42]: You can use our `useModal` component imported from `@refinedev/core` instead of use `useState`

**Review by salihozdemir — APPROVED** [review]

**Pull request merged**